src/jit/morph.cpp

```diff
@@ -12752,39 +12752,51 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 {
                     assert(temp->OperIsLocal());
 
-                    LclVarDsc* varDsc = &(lvaTable[temp->AsLclVarCommon()->gtLclNum]);
-                    // Make sure we don't separately promote the fields of this struct.
-                    if (varDsc->lvRegStruct)
-                    {
-                        // We can enregister, but can't promote.
-                        varDsc->lvPromoted = false;
-                    }
-                    else
-                    {
-                        lvaSetVarDoNotEnregister(temp->gtLclVarCommon.gtLclNum DEBUGARG(DNER_LocalField));
-                    }
+                    const unsigned   lclNum = temp->AsLclVarCommon()->gtLclNum;
+                    LclVarDsc* const varDsc = &lvaTable[lclNum];
 
-                    // We will turn a GT_LCL_VAR into a GT_LCL_FLD with an gtLclOffs of 'ival'
-                    // or if we already have a GT_LCL_FLD we will adjust the gtLclOffs by adding 'ival'
-                    // Then we change the type of the GT_LCL_FLD to match the orginal GT_IND type.
-                    //
-                    if (temp->OperGet() == GT_LCL_FLD)
+                    // If the size of the load is greater than the size of the lclVar, we cannot fold this access into
+                    // a lclFld: the access represented by an lclFld node must begin at or after the start of the
+                    // lclVar and must not extend beyond the end of the lclVar.
+                    if (varDsc->lvExactSize < genTypeSize(typ))
```

Comment:
Minor: Based on the comment above (".. greater than.."), one would expect to see a > condition below.  It would make it easier to follow the code by changing the if condition as follows.

`if (genTypeSize(typ) > varDsc->lvExactSize)`